digital-workspace/latest/develop/extension-mechanism.md

```diff
@@ -0,0 +1,236 @@
+---
+title: Extension mechanism
+---
+
+The Alfresco Development Framework extension mechanism allows you to add, remove, and change the behavior of your applications. The extension mechanism supports the Digital Workspace, and is the recommended way to make direct changes to its source code.
+Customizations that are implemented through the extension mechanism are more maintainable because the changes are modular and isolated from the core of the application. This approach to customizing the Digital Workspace ensures upgrades are easier to manage and test.
+
+Use the Alfresco Digital Workspace and develop an extension if your use case is mainly covered by the Alfresco Digital Workspace, but some changes are required to meet the expectations and the requirements. In case of complex changes, a custom ADF based application is an alternative.
+
+## How to create an extension for the Digital Workspace
+
+Use this this information to develop a `hello word` extension for the Digital Workspace.
+
+### Prerequisites and requirements
+
+* Alfresco Content Services - Enterprise Edition.
+  * Open your browser and check everything starts up correctly:
+
+  ```bash
+  Alfresco: http://localhost:8080/alfresco
+  ```
+
+* The latest **LTS** version of `Node.js`.
+
+### Create a Digital Workspace extension
+
+Use the [Nx](https://nx.dev/) developer tools for monorepos to create the Digital Workspace extension.
+
+1. Install `nx` cli globally.
+
+    `npm install -g nx`
+
+2. Use the following command to create a new extension called `my-extension`, from the root directory of the Digital Workspace project.
+
```

Comment:
It should be required to add a link to document section describing where to obtain and configure a Digital Workspace project.

Reply:
See above